test/k8s-integration/version_test.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestParseVersion(t *testing.T) {
+	tests := []struct {
+		version   string
+		expectErr bool
+		expectedV Version
+	}{
+		// Positive test cases.
+		{
+			version: "v1.18.0",
+			expectedV: Version{
+				version: [4]int{1, 18, 0, -1},
+			},
+		},
+		{
+			version: "v1.18.0-gke.0",
+			expectedV: Version{
+				version: [4]int{1, 18, 0, 0},
+			},
+		},
+		{
+			version: "1.18.0",
+			expectedV: Version{
+				version: [4]int{1, 18, 0, -1},
+			},
+		},
+		{
+			version: "1.18.0-gke.0",
+			expectedV: Version{
+				version: [4]int{1, 18, 0, 0},
+			},
+		},
+		{
+			version: "1.18.3-gke.10",
+			expectedV: Version{
+				version: [4]int{1, 18, 3, 10},
+			},
+		},
+		{
+			version: "1.18.9",
+			expectedV: Version{
+				version: [4]int{1, 18, 9, -1},
+			},
+		},
+		{
+			version: "1.18.10-gke.10",
+			expectedV: Version{
+				version: [4]int{1, 18, 10, 10},
+			},
+		},
+		{
+			version: "10.18.10-gke.10",
+			expectedV: Version{
+				version: [4]int{10, 18, 10, 10},
+			},
+		},
+		// Negative test cases
+		{
+			version:   "-1.18.9",
+			expectErr: true,
+		},
+		{
+			version:   "1.-18.9",
+			expectErr: true,
+		},
+		{
+			version:   "1.18",
+			expectErr: true,
+		},
+	}
+
+	fmt.Println("Num test cases:", len(tests))
+	for i, tc := range tests {
+		fmt.Printf("Run test: %s, test case number:%d\n", t.Name(), i)
+		gotV, err := parseVersion(tc.version)
+		if err != nil {
+			if !tc.expectErr {
+				t.Fatalf("Got unexpected err: %v", err)
+			}
+			continue
+		}
+
+		if err == nil && tc.expectErr {
+			t.Fatal("Got no error but expected one")
+			continue
+		}
+
+		if gotV.version[0] != tc.expectedV.version[0] ||
+			gotV.version[1] != tc.expectedV.version[1] ||
+			gotV.version[2] != tc.expectedV.version[2] ||
+			gotV.version[3] != tc.expectedV.version[3] {
+			t.Fatalf("Got version: %s, expected: %s", gotV.String(), tc.expectedV.String())
+		}
+	}
+}
+
+func TestIsVersionLessThan(t *testing.T) {
+	tests := []struct {
+		leftVersion  string
+		rightVersion string
+		expectRes    bool
+	}{
+		// Positive cases.
```

Comment:
Should cover left > right as well.

Also include a case like "1.18.0" and "1.18.1" where both versions lack the gke part and they are unequal

Reply:
Done